util/parsec_file.py

```
@@ -33,15 +33,17 @@ class ParsecResult:
 
-    # The warning suppression below is required; nx.DiGraph does not expose subscriptable types.
+    # MDE: There should be a comment stating the type of the nodes in `call_graph`, even if the
+    # type-checker cannot utilize it.
+    # "ignore[type-arg]" because nx.DiGraph does not expose subscriptable types.
     call_graph: nx.DiGraph  # type: ignore[type-arg]
     file_path: Path
     enums: list[Any] = field(default_factory=list)
     files: list[str] = field(default_factory=list)
     functions: dict[str, ParsecFunction] = field(default_factory=dict)
 
     def __init__(self, file_path: Path):
```

Ruff (0.14.8)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)